src/Http/Routing/src/RouteOptions.cs

```diff
@@ -9,7 +9,22 @@ namespace Microsoft.AspNetCore.Routing
 {
     public class RouteOptions
     {
-        public ICollection<EndpointDataSource> EndpointDataSources { get; internal set; }
+        /// <summary>
+        /// Gets a collection of <see cref="EndpointDataSource"/> instances configured with routing.
+        /// </summary>
+        public ICollection<EndpointDataSource> EndpointDataSources
```

Comment:
Does it make sense to think about different fixes for this? (or even making this internal/separate).

I'm not sure that *fixing* this behaviour will fix a real problem. I think if you try to use the `UseEndpoints` and this you're going to have a bad time regardless.

I'm not totally certain that we have a use case for users to interact with this property.

Reply:
UseRouting won't use these, but anyone else integrating with endpoints will, e.g. link generation.

If we don't make this modifiable then UseEndpoints would be the only way to specify a data source.

Reply:
> If you want to write to this property, that will be useful because ??????

We don't have any scenarios or documentation that use this. It literally has not come up once. I'm pretty lukewarm about public API for things that we wouldn't tell people to do.

Reply:
Fair enough. It was added to replace registering endpoint data sources via DI which I loathed, but we have UseEndpoints now.

I'll make it internal and see whether any problems come up.